DEV_GUIDE.md

```diff
@@ -0,0 +1,39 @@
+Formplayer Dev Resources
+========================
+
+This document is aimed at developers starting out with Formplayer and
+particularly those who may not be familiar with Java or Spring. It lays out a set of
+resources that can be used as to learn about the technologies used.
+
+Formplayer is a [Spring Boot](https://docs.spring.io/spring-boot/docs/current/reference/html/index.html) which
+takes an opinionated view of the Spring platform and third-party libraries,
+so that you can get started with minimum fuss. Most Spring Boot applications need very
+little Spring configuration.
+
+[Spring](https://spring.io/) is a collection of tools and libraries that have been
+built to make developing Java applications easier and safer. At it's core is the Spring Framework
+which provides the technologies that support the rest of the tools: dependency injection, events, resources, i18n,
+validation, data binding, type conversion, SpEL (Spring Expressive Language), AOP (Aspect Oriented Programming)
+
+Developing Formplayer will require some level of knowledge of at least the following core components:
+* [dependency injection (IoC)](https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#beans)
+* [AOP](https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#aop)
+
+Spring Boot ties a number of the components of Spring together in way to meets the needs of most projects. It does
+this by using [conventions](boot_conventions), but it does not stop you from overriding the conventions.
```

Comment:
don't think link for `conventions` is working here. it's getting resolved to https://github.com/dimagi/formplayer/blob/master/boot_conventions

Reply:
thanks fixed https://github.com/dimagi/formplayer/commit/3c7db7c71f70aad0dac3c740bd77861649dd2405